src/backend/marsha/core/serializers.py

```diff
@@ -192,6 +198,50 @@ def get_url(self, obj):
             return url
         return None
 
+    def get_url(self, obj):
+        """Url of the timed text track, signed with a CloudFront key if activated.
+
+        Parameters
+        ----------
+        obj : Type[models.TimedTextTrack]
+            The timed text track that we want to serialize
+
+        Returns
+        -------
+        string or None
+            The url for the timed text track converted to vtt.
+            None if the timed text track is still not uploaded to S3 with success.
+
+        """
+        if obj.uploaded_on:
+
+            base = "{protocol:s}://{cloudfront:s}/{video!s}".format(
+                protocol=settings.AWS_S3_URL_PROTOCOL,
+                cloudfront=settings.CLOUDFRONT_DOMAIN,
+                video=obj.video.pk,
+            )
+            stamp = time_utils.to_timestamp(obj.uploaded_on)
+            url = "{base:s}/timedtext/{stamp:s}_{language:s}{mode:s}.vtt".format(
+                base=base,
+                stamp=stamp,
+                language=obj.language,
+                mode="_{:s}".format(obj.mode) if obj.mode else "",
+            )
+
+            # Sign the url only if the functionality is activated
+            if settings.CLOUDFRONT_SIGNED_URLS_ACTIVE:
+                date_less_than = timezone.now() + timedelta(
+                    seconds=settings.CLOUDFRONT_SIGNED_URLS_VALIDITY
+                )
+                cloudfront_signer = CloudFrontSigner(
+                    settings.CLOUDFRONT_ACCESS_KEY_ID, cloudfront_utils.rsa_signer
+                )
+                url = cloudfront_signer.generate_presigned_url(
+                    url, date_less_than=date_less_than
+                )
+            return url
+        return None
```

Comment:
Could be mutualized with the other method?

Reply:
Yes :+1: